src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -401,6 +407,30 @@ private void AddCredentialToCache(Uri serverUri, ulong authAvail, NetworkCredent
             }
         }
 
+        private void AddResponseCookies(Uri serverUri, HttpResponseMessage response)
+        {
+            if (!_useCookie)
```

Comment:
Actually in thinking about this some bug, this code is not really a bug, per se, because in the `HttpClientHandler` layer we do NOT expose the `UseInternalCookieStoreOnly` option.  There is no way to set it or get it.  We did this to preserve app-compat when using HttpClientHandler directly.  So, the code is valid.

Developers on Windows that want a higher performance cookie store (internal cookie store in WinHttp) can get that by creating the WinHttpHandler directly and using the specific `CookieUsePolicy` enums.